gdal/frmts/netcdf/netcdfdataset.cpp

```diff
@@ -4470,39 +4480,66 @@ int NCDFWriteSRSVariable(int cdfid, const OGRSpatialReference* poSRS,
     return NCDFVarID;
 }
 
+// Use a default name (variable name: crs)
+int NCDFWriteSRSVariable(int cdfid, const OGRSpatialReference* poSRS,
+                                char **ppszCFProjection, bool bWriteGDALTags)
+{
+    std::string srsVarName = ""; // for versions NOT CF-1.8
+
+    return NCDFWriteSRSVariable(cdfid, poSRS, ppszCFProjection,
+                                bWriteGDALTags, srsVarName);
+}
+
 /************************************************************************/
 /*                   NCDFWriteLonLatVarsAttributes()                    */
 /************************************************************************/
 
-void NCDFWriteLonLatVarsAttributes(int cdfid, int nVarLonID, int nVarLatID)
+void NCDFWriteLonLatVarsAttributes(int cdfid, int nVarLonID, int nVarLatID, nccfdriver::netCDFVID * vcdf)
 {
-    int status =
-        nc_put_att_text(cdfid, nVarLatID, CF_STD_NAME,
-                        strlen(CF_LATITUDE_STD_NAME), CF_LATITUDE_STD_NAME);
-    NCDF_ERR(status);
+    if(vcdf == nullptr)
```

Comment:
I also found the amount of conditional checks for using "nc_put_vatt"

But as far as I know, in netCDF there is a total ordering on the creations of Dimension, Variables, and Attributes of Variables- that is, first you must define Dimensions for a variable then the Variables themselves, and then finally, the Attributes. Unfortunately since attributes depend on variables and variables depend on dimensions, using nc_put_att_* would not work due to the fact that the dimensions are not written until the very end of translation.

Reply:
I don't really understand your answer here; What I mean is that
`            nc_put_att_text(cdfid, nVarLatID, CF_STD_NAME,
                            strlen(CF_LATITUDE_STD_NAME), CF_LATITUDE_STD_NAME);
`
could very well be implemented by
```
vcdf->nc_put_vatt_text(nVarLatID, CF_STD_NAME, CF_LATITUDE_STD_NAME);
```

Reply:
Oh. The reason why `nc_put_att_*` is still there is that the WKT implementation hasn't been modified to use the "virtual ID" (though I think with some more declarations it could be). In fact the reason why I called it "virtual ID" instead virtual dataset or whatever was so it could be used with groups in WKT datasets.

What I was trying to say above is that buffering is done because the variable itself isn't yet resident in the netCDF file- it's only in the netCDFVID at the moment so implementing nc_put_vatt_* in this manner would result in an error.

Reply:
what I meant is that netCDFVID should be an abstract class, with 2 implementations: the one you did with this buffering, and another one that just calls directly the nc_XXXX() functions

Reply:
Ooh hmm that could be a neat solution. Sure, I'll go look into it for this and the rest below.

Reply:
As a small note: I decided to just implement this through a flag in the existing netCDFVID class instead of making a small class hierarchy in order to not make things too complicated (since the full netCDFVID provides some additional features not provided by the netCDF Library).